radareorg/cutter
# Cmake install

**Detailed description**
First steps towards CMake install (maybe in combination with cpack) being usable for packaging Cutter.

**Test plan (required)**
* make sure compilation on old Ubuntu still works, CMake install support is optional :heavy_check_mark: 

Test install on Windows using Bundled radare2. ✔️ 
* Build using CMake and bundled r2
* test that Cutter in Build directory opens and r2 resources (color themes, greetings, ...) are loaded
* repeat previous step in install directory (requiring manual Qt copying is acceptable)

 Test make install on Linux, recommended doing in a VM. :heavy_check_mark:*

* install r2
* build Cutter using system r2
* `sudo make install`
* make sure executable is installed in correct place
* make sure Cutter executable opens and there are no dynamic library loading errors
* make sure r2 resources  (color themes, greeting messages, ..) are found
* make sure Desktop environment finds the .desktop file, Cutter shows up in list of programs with correct icon
* make sure the icon gets used when Cutter executable is running
* test that installed Cutter dev files can be used for compiling plugins by compiling Cutter Ghidra plugin externally 

\* Compiling r2ghidra requires fresh CMake so the plugin compilation part couldn't be tested with Ubuntu 18.04 default CMake. It was also necessary to remove FindCMake because find_package prefers FindCMake.cmake over the installed cmake config unless CMAKE_FIND_PACKAGE_PREFER_CONFIG option is used. GHidra plugin will need to be updated, the old mechanism can't be removed until official releases are built using new mechanism.

**Known limitations**
When using with CUTTER_USE_BUNDLED_RADARE2=ON doing `make DESTDIR=foo install` directly instead of `make` and `make install separately` affects r2 meson output directories thus breaking the build.

**Follow up tasks**
* translation installation
* Python bindings and Python file installation for Windows, Appimage, macOS packages
* Bundled plugin installation (r2ghidra, r2-dec)
* macOS file installation
* ~plugin compilation on windows~ export annotation

**Closing issues**

closes #1896 
closes #1914
